lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1147,27 +1173,27 @@ private function isLimitQuery() : bool
      */
     private function getSQLForInsert() : string
     {
-        return 'INSERT INTO ' . $this->sqlParts['table'] .
-        ' (' . implode(', ', array_keys($this->sqlParts['values'])) . ')' .
-        ' VALUES(' . implode(', ', $this->sqlParts['values']) . ')';
+        return 'INSERT INTO ' . $this->table .
+        ' (' . implode(', ', array_keys($this->values)) . ')' .
+        ' VALUES(' . implode(', ', $this->values) . ')';
     }
 
     /**
      * Converts this instance into an UPDATE string in SQL.
      */
     private function getSQLForUpdate() : string
     {
-        return 'UPDATE ' . $this->sqlParts['table']
-            . ' SET ' . implode(', ', $this->sqlParts['set'])
-            . ($this->sqlParts['where'] !== null ? ' WHERE ' . ((string) $this->sqlParts['where']) : '');
+        return 'UPDATE ' . $this->table
+            . ' SET ' . implode(', ', $this->set)
+            . ($this->where !== null ? ' WHERE ' . ((string) $this->where) : '');
     }
 
     /**
      * Converts this instance into a DELETE string in SQL.
      */
     private function getSQLForDelete() : string
     {
-        return 'DELETE FROM ' . $this->sqlParts['table'] . ($this->sqlParts['where'] !== null ? ' WHERE ' . ((string) $this->sqlParts['where']) : '');
+        return 'DELETE FROM ' . $this->table . ($this->where !== null ? ' WHERE ' . ((string) $this->where) : '');
```

Comment:
Please use an `if` and remove `(string)` if possible.

Reply:
Done in 08e9edf9fddeb4883dbb7186a438c93611fcc4b7.